katcp/test/test_server.py

```diff
@@ -1014,13 +1014,26 @@ def test_sampling(self):
         self.server.ioloop.add_callback(self.server.clear_strategies, client_conn)
         self.server.sync_with_ioloop()
         self.client.assert_request_succeeds("sensor-sampling", "an.int",
-                                            args_equal=[b"an.int", b"none"])
-
+                                            args_equal=["an.int", "none"])
         # Check that we did not accidentally clobber the strategy datastructure
         # in the proccess
         self.client.assert_request_succeeds(
             "sensor-sampling", "an.int", "period", 0.125)
 
+    def test_assert_request_succeeds(self):
+        """Test exercises assert_request_succeeds"""
+        self.client.assert_request_succeeds("sensor-sampling", "an.int",
+                                            args_equal=["an.int", "none"])
+        self.client.assert_request_succeeds("sensor-sampling", b"an.int",
+                                            args_equal=["an.int", "none"])
+        self.client.assert_request_succeeds("sensor-sampling", b"an.int",
+                                            args_equal=[b"an.int", b"none"])
+        self.client.assert_request_succeeds("log-level", "debug", args_equal=["debug"],
+                                            args_echo=True)
+        self.client.assert_request_succeeds("log-level", "debug", args_equal=[b"debug"],
+                                            args_echo=True)
+        levels = ['off', b'fatal', 'error', b'warn', 'info', 'debug', 'trace', 'all']
+        self.client.assert_request_succeeds("log-level", args_in=[[level] for level in levels])
 
```

Comment:
Will tests still be added for these?
- `informs_args_equal` with byte strings and native strings
- testing that an exception is raised when `assert_request_succeeds` is used incorrectly - e.g., using it to call `?raise-fail`.

Reply:
Yes this will be added.

Reply:
@ajoubertza Should I add an exception that is raised when `assert_request_succeeds` is used incorrectly and add a test for that exception?

Reply:
Not sure if I understand the question.  Are you asking if we should add a new type of exception for this case?  Probably not.  An `AssertionError` seems like a reasonable choice for a test method called `assert_something`.  What does it raise?

Reply:
> Ok, I was asking if I should throw exceptions in the assert_request_suceeds as there is it throws no exceptions. It raises nothing at the moment other than the exceptions you will encounter in assertEqual

No new exception to be added - the exception raised when `assertEqual` fails is perfect.  Or if we used a non-existent request, it would fail on this [`assertTrue`](https://github.com/ska-sa/katcp-python/blob/6ba34f3832ab8386546b4cdda5bff8d33d6ddbbc/katcp/testutils.py#L784).  I see you have added tests like that, which is what I was suggesting.